src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/ExternalAccountManager.java

```diff
@@ -0,0 +1,143 @@
+                            //    Expect: DELIVERY_FAILED, but non-null "mailjet_id"
+                            //    Action: same as deletion? Or just remove from lists for now?
+                            log.debug("Case: delivery failed.");
+                            mailjetApi.updateUserSubscriptions(mailjetId, MailJetSubscriptionAction.REMOVE, MailJetSubscriptionAction.REMOVE);
+                        } else if (!accountEmail.equals(mailjetDetails.getString("Email"))) {
+                            // Case: account email change:
+                            //    Expect: non-null "mailjet_id", email in MailJet != email in database
+                            //    Action: delete old email, add new user for new email
+                            log.debug("Case: account email change.");
+                            mailjetApi.permanentlyDeleteAccountById(mailjetId);
+                            mailjetId = mailjetApi.addNewUser(accountEmail);
+                            updateUserOnMailJet(mailjetId, userRecord);
+                        } else {
+                            // Case: changed details/preferences:
+                            //    Expect: not deleted, not DELIVERY_FAILED
+                            //    Action: update details, update subscriptions, update provider_last_updated
+                            log.debug("Case: generic preferences update.");
+                            updateUserOnMailJet(mailjetId, userRecord);
+                        }
+                    } else {
+                        if (!EmailVerificationStatus.DELIVERY_FAILED.equals(userRecord.getEmailVerificationStatus())) {
+                            // Case: new to Isaac, not on MailJet:
+                            //    Expect: null "mailjet_id", not DELIVERY_FAILED
+                            //    Action: create MailJet ID, update details, update subscriptions, update provider_last_updated
+                            log.debug("Case: new to Isaac/not on MailJet");
+                            mailjetId = mailjetApi.addNewUser(accountEmail);
+                            updateUserOnMailJet(mailjetId, userRecord);
+                        } else {
+                            // Not on MailJet, but invalid email so don't add to MailJet.
+                            // Do we need this row?
+                            log.debug("Case: invalid/incorrect user to skip.");
+                            database.updateExternalAccount(userId, null);
+                        }
+                    }
+                    // Iff action done successfully, update the provider_last_updated time:
+                    log.debug("Update provider_last_updated.");
+                    database.updateProviderLastUpdated(userId);
+
+                } catch (SegueDatabaseException e) {
+                    log.error(String.format("Error storing record of MailJet update to user (%s)!", userId));
```

Comment:
All this code is asynchronous and if the last updated isn't modified will just run again the next time the sync is triggered.
Creation and updating are idempotent and so safe, I should check about deletion since that may throw an error if attempted twice.